src/daemon/mod.rs

```diff
@@ -78,13 +83,19 @@ pub fn ipc_shmem_conf() -> ShmemConf {
 }
 
 fn unblock_parent_process() -> anyhow::Result<()> {
-    let shmem = ipc_shmem_conf().open()?;
-    let (event, _) =
-        unsafe { Event::from_existing(shmem.as_ptr()).map_err(|err| anyhow::anyhow!("{err}")) }?;
-
-    event
-        .set(EventState::Signaled)
-        .map_err(|err| anyhow::anyhow!("{err}"))
+    static UNBLOCKED: AtomicBool = AtomicBool::new(false);
```

Comment:
Should we consider deprecating the detach mode? It requires additional logic but does not provide irreplaceable value. @LesnyRumcajs

Reply:
I'm okay with this, provided we have a clear migration document for using background tasks. It's a breaking change that will likely affect at least few services out there, including our own CI (perhaps sync check as well).